k-distribution/k-tutorial/1_basic/02_basics/README.md

```diff
@@ -1 +1,214 @@
+call to `kompile` to execute this program.
+
+You will get the following output:
+
+```
+<k>
+  Yellow ( ) ~> .
+</k>
+```
+
+For now, don't worry about the `<k>`, `</k>`, or ` ~> .` portions of this
+output file. Instead, let's look at what this definition and program did.
+
+## Productions, Constructors, and Functions
+
+The first thing to realize is that this K definition contains 5 productions.
+Productions are introduced with the **syntax** keyword, followed by a **sort**,
+followed by the operator `::=` followed by the definition of one or more
+productions themselves, separated by the `|` operator. There are different
+types of productions, but for now we only care about **constructors** and 
+**functions**. Each declaration separated by the `|` operator is individually
+a single production, and the `|` symbol simply groups together productions that
+have the same sort. For example, we could equally have written an identical K
+definition like so:
+
+```k
+module LESSON-02-B
+
+  syntax Color ::= Yellow()
+  syntax Color ::= Blue()
+  syntax Fruit ::= Banana()
+  syntax Fruit ::= Blueberry()
+  syntax Color ::= colorOf(Fruit) [function]
+
+  rule colorOf(Banana()) => Yellow()
+  rule colorOf(Blueberry()) => Blue()
+
+endmodule
+```
+
+Or even:
+
+```k
+module LESSON-02-C
+
+  syntax Color ::= Yellow()
+                 | Blue()
+                 | colorOf(Fruit) [function]
+  syntax Fruit ::= Banana()
+                 | Blueberry()
+
+  rule colorOf(Banana()) => Yellow()
+  rule colorOf(Blueberry()) => Blue()
+
+endmodule
+```
+
+Each of these types of productions named above has the same underlying syntax,
+but context and **attributes** are used to distinguish between the different
+types. Tokens, brackets, lists, macros, aliases, and anywhere productions will
+be covered in a later lesson, but this lesson does introduce us to constructors
+and functions. `Yellow()`, `Blue()`, `Banana()`, and `Blueberry()` are
+constructors. You can think of a constructor like a constructor for an
+algebraic data type, if you're familiar with a functional language. The data
+type itself is the sort that appears on the left of the `::=` operator. Sorts
+in K consist of uppercase identifiers.
+
+Constructors can have arguments, but these ones do not. We will cover the
+syntax of productions in detail in the next lesson, but for now, you can write
+a production with no arguments as an uppercase or lowercase identifier followed
+by the `()` operator.
+
+A function is distinguished from a constructor by the presence of the
+**function** attribute. Attributes appear in a comma separated list between
+square brackets after any sentence, including both productions and rules.
+Various attributes with built-in meanings exist in K and will be discussed
+throughout the tutorial, but the user can also give any sentence a user-defined
+attribute using the same syntax by simply using an identifier for the attribute
+which is not reserved by K.
```

Comment:
```suggestion
Various attributes with built-in meanings exist in K and will be discussed
throughout the tutorial.
```

No need to mention user-defined attributes yet, not at least until they learn about syntax priorities. I would also like to minimize mention of this feature, as it's more controversial than other features (some would like to see it removed, including myself).